packages/backend/src/utils/progressiveEvent.ts

```diff
@@ -0,0 +1,50 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface ProgressiveEvent {
+  id: string;
+  status: 'error' | 'completed' | 'progress' | 'canceled';
+  message?: string;
+}
+
+export interface CompletionProgressiveEvent extends ProgressiveEvent {
+  status: 'completed' | 'error' | 'canceled';
+  duration: number;
+}
+
+export interface ProgressProgressiveEvent extends ProgressiveEvent {
```

Comment:
ProgressProgressiveEvent ? Why such wording ?

Reply:
This way it's self-explanatory if you read it in some other code. If you just read ProgressEvent, you need to check what it extends. ProgressEvent looks to me too generic but i'm open to suggestions